components/formats-bsd/src/loci/formats/in/TileJPEGReader.java

```diff
@@ -122,34 +105,24 @@ public void initFile(String id) throws FormatException, IOException {
     super.initFile(id);
 
     in = new RandomAccessInputStream(id);
-    decoder = new JPEGTileDecoder();
-    decoder.initialize(in, 0, 1, 0);
+    JPEGTileDecoder decoder = new JPEGTileDecoder();
+    int[] dimensions = decoder.preprocess(in);
+    decoder.close();
```

Comment:
The integration with the work in #2883 makes complete sense to drive the new low-level codec component. Would the following step here be to review all decoders which should be made `Closeable`, release version `0.2.0` and propagate upstream?